sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java

```diff
@@ -22,11 +22,28 @@
 import java.util.Locale;
 import java.util.Objects;
 
-/*
+/**
  * A class for aggregating EventData into a single, size-limited, batch that will be treated as a single message when
  * sent to the Azure Event Hubs service.
+ *
+ * <p><strong>Publish events using an {@link EventDataBatch}.</strong></p>
+ *
+ * Developers can create {@link EventDataBatch batches}, add the events they want in that batch, and publish these
+ * events together. When creating a {@link EventDataBatch batch}, developers can specify a set of {@link BatchOptions
+ * options} to configure this batch.
+ *
+ * In the scenario below, the developer is creating a networked video game. They want to receive telemetry about their
+ * users' gaming systems, but do not want to slow down the network with telemetry. So they limit the size of their
+ * {@link EventDataBatch batches} to be no larger than 256 bytes. The events within the batch also get hashed to the
+ * same partition because they all share the same {@link BatchOptions#partitionKey()}.
+ *
+ * {@codesnippet com.azure.messaging.eventhubs.eventhubproducer.send#eventdatabatch}
```

Comment:
You should determine if it is better to reproduce the sample code snippet from event hub producer (as is done here), or if you should refer the user to the class 'for further code samples'.

Reply:
Duplicated code is hard to maintain. I removed it and replaced it with a `@see.`